src/text/encoder.rs

```diff
@@ -0,0 +1,264 @@
+use crate::result::IonResult;
+use crate::IonType;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, FixedOffset};
+use std::io;
+
+/// Utility type for writing scalar Rust values as Ion text.
+pub struct TextEncoder {}
```

Comment:
The `TextEncoder` type can encode any scalar value to an `io::Write` sink. `TextWriter` wraps a `TextEncoder`; it provides a buffer and manages container/annotation state.

Reply:
But why do we need both?  `TextEncoder` if I squint is really just some parts of `TextWriter` factored in a way to make it useful in other contexts.  What would that context be other than `TextWriter`?  In other words what is the separation buying us other than splitting some of the encoding concerns?

I think that intent should be captured in the doc comment or somewhere, because it did confuse me.